observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerDeploymentProcessor.java

```diff
@@ -21,32 +18,29 @@
 import org.jboss.as.server.deployment.DeploymentUnitProcessor;
 import org.jboss.as.weld.WeldCapability;
 import org.wildfly.extension.micrometer.api.MicrometerCdiExtension;
-import org.wildfly.extension.micrometer.registry.WildFlyRegistry;
+import org.wildfly.subsystem.service.ServiceInstaller;
 
 class MicrometerDeploymentProcessor implements DeploymentUnitProcessor {
-    private final boolean exposeAnySubsystem;
-    private final List<String> exposedSubsystems;
-    private final Supplier<WildFlyRegistry> registrySupplier;
+    static final String WELD_CAPABILITY_NAME = "org.wildfly.weld";
+
+    private final MicrometerSubsystemRegistrar.MicrometerDeploymentConfiguration config;
 
-    MicrometerDeploymentProcessor(boolean exposeAnySubsystem,
-                                  List<String> exposedSubsystems,
-                                  Supplier<WildFlyRegistry> registrySupplier) {
-        this.exposeAnySubsystem = exposeAnySubsystem;
-        this.exposedSubsystems = exposedSubsystems;
-        this.registrySupplier = registrySupplier;
+    MicrometerDeploymentProcessor(MicrometerSubsystemRegistrar.MicrometerDeploymentConfiguration config) {
+        this.config = config;
     }
 
     @Override
     public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
         final DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
 
-        MicrometerDeploymentService.install(deploymentPhaseContext.getServiceTarget(),
-                deploymentPhaseContext,
-                deploymentUnit.getAttachment(DeploymentModelUtils.DEPLOYMENT_RESOURCE),
-                deploymentUnit.getAttachment(DeploymentModelUtils.MUTABLE_REGISTRATION_ATTACHMENT),
-                registrySupplier,
-                exposeAnySubsystem,
-                exposedSubsystems);
+        ServiceInstaller.builder(() -> new MicrometerDeploymentService(
+                        deploymentUnit.getAttachment(DeploymentModelUtils.DEPLOYMENT_RESOURCE),
+                        deploymentUnit.getAttachment(DeploymentModelUtils.MUTABLE_REGISTRATION_ATTACHMENT),
+                        createDeploymentAddressPrefix(deploymentUnit),
+                        config.getCollectorSupplier(),
+                        config.getSubsystemFilter()
+                )).asActive()
+                .build();
```

Comment:
I think you are missing `.install(deploymentPhaseContext)`? :)

Reply:
Probably. :P Fixed.